src/analyses/natural_loops.h

```diff
@@ -7,7 +7,17 @@ Author: Georg Weissenbacher, georg@weissenbacher.name
 \*******************************************************************/
 
 /// \file
-/// Compute natural loops in a goto_function
+/// Compute natural loops in a goto_function.
+///
+/// A natural loop is when a graphs nodes make one self-encapsulating circle
+/// with no incoming edges from external nodes. This is to say that
+/// A -> B -> C -> D -> A is a natural loop, but if B has an incoming edge from X,
+/// then it isn't a natural loop, because X is an external node. Outgoing edges don't
+/// affect the natural-ness of a loop.
+///
+/// The detection to work out if a nodes children can only be accessed through itself,
```

Comment:
I might be tempted to invert the order of this paragraph a bit, and say something like "\ref cfg_dominators_templatet is used to provide the dominator analysis that is used to determine if a node is reachable only through itself, and is thus part of a natural loop." ?